### DISCUSSION TITLE
How to make a workflow_call conditional on if files changed in a directory?

### DISCUSSION THREAD

**Discussion by damies13** [community/community#133797] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi All,

My workflow action currently does this (main.yaml):
- run a job to create a python package (package.yaml)
- run tests in parallel
  - run tests for component A (tests_a.yaml) (needs package.yaml)
      - runs a 3 x 5 matrix (platform and python versions)
  - run tests for component M (tests_m.yaml) (needs package.yaml)
      - runs a 3 x 5 matrix (platform and python versions)
  - run tests for component R (tests_r.yaml) (needs package.yaml)
      - runs a 3 x 5 matrix (platform and python versions)
- if all tests pass and is release branch with all issues closed, then publish the python package (publish.yaml)

What I would like to do is change the either main.yaml or tests_*.yaml to apply a filter so the tests_*.yaml only runs their jobs if there were changes related to that component (either in the component's code or in the component's test cases)

What I've looked at so far and doesn't work for my situation:
- [Using filters to target specific paths for pull request or push events](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-filters-to-target-specific-paths-for-pull-request-or-push-events) > [Example: Including paths](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#example-including-paths)
unfortunately this only works with  `on push`, which is too early for my workflow, I need to apply the filter to which test groups to run after the package is created, not before
  - Packaging the components separately with separate workflows using the paths filter on push causes other issues with the version numbers and publishing, so this is not an option for me
- I did try applying the `paths` filter to `workflow_run` but got the error `The workflow is not valid`, this would have been the best solution unfortunately it doesn't work
```
on:
  workflow_call:
    paths:
      - 'component_a/**'
      - 'Tests/Regression/component a/**'
```
- `dorny/paths-filter@v3` and similar actions - these happen in the job steps so too late in the workflow
  - main.yml doesn't have any steps, so if I added this here it would require the workflow to start an extra runner to get the answer. I also found that I any variables I set in the package.yaml do not get passed back to main.yaml or the other workflows, otherwise this could have been an option
  - in tests_*.yaml, I want to apply the filter before the jobs section, doing it within the steps of a job would be too late as the workflow engine still has to spin up the 15 runners to get to the step to find it didn't need the runner (this is what I'm trying to avoid)

Any suggestions?

Dave.

**Comment by damies13** [comment, marked as answer]
I found my answer.

This link was the starting point: https://www.meziantou.net/executing-github-actions-jobs-or-steps-only-when-specific-files-change.htm

I had to define the output from the job, then use that to define the output from the workflow, then the parent workflow (main.yaml) could get the output from the child workflow and use it in an if statement.

So I ended up adding `dorny/paths-filter@v3` steps to my package workflow and setting output values for run_a, run_m, and run_r

**Comment by manikandanvios** [comment]
Hi All,
  I have do create github actions flutter ios but i got error while run

      - name: Install dependencies
        run: flutter pub get

Error -
  git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights

i got error like this please help for resolve this issue.

**Reply by Psc544** [reply]
https://gitlab.com/mailman/hyperkitty.git
